src/utils.ts

```diff
@@ -227,3 +228,51 @@ export function arrify<T>(val: T | T[]) {
 
   return Array.isArray(val) ? val : [val];
 }
+
+export function ensureProgram(instance: TSInstance) {
```

Comment:
I'm not confident I understood the purpose of this function correctly. I moved it to utils because it seemed it was a way to get the `typescript.Program` out of the watch or the language service, but then noticed it potentially has side effects—I'm not sure what `watchHost.updateRootFileNames()` does and if it's harmful to call it more often than it was previously.

Reply:
Since it's only called conditionally on `instance.changedFilesList` then hopefully behaviour would be unchanged?